### DISCUSSION TITLE
Controlling GraphCanvas Size Within Parent Container

### DISCUSSION THREAD

**Discussion by rb-x** [reaviz/reagraph#138] [Q&A] [answered]
I'm using a GraphCanvas component within a parent container, and I'm facing difficulties in controlling the size of the GraphCanvas. I want to either specify relative (percentage-based) or absolute (fixed pixel) width and height for the GraphCanvas. Currently, it takes up the entire window's space, and it even overlaps my side menu, and my attempts to apply CSS rules to the canvas element have not been successful.

Here's how I'm trying to set the size:

```jsx
<div style={{ width: '50%', height: '50%' }}>
    <GraphCanvas nodes={nodes} edges={edges} />
</div>
```

I'm looking for a solution or guidance on how to make the GraphCanvas either match the size of its parent node or specify its width and height effectively via props maybe ?

**Comment by rickypaques** [comment]
Hi,
you can inspect element of the canvas and find this className (on image)
![image](https://github.com/reaviz/reagraph/assets/142203934/5ecfa538-264f-4112-b825-2e1f8083e919)

you can manipulation by DOM to get the className, set height and position you want.

**Comment by thobhanifreddy** [comment]
make position relative for parent div

```
<div style={{ position: "relative", width: '50%', height: '50%' }}>
    <GraphCanvas nodes={nodes} edges={edges} />
</div>
```

**Reply by wilkyrlx** [reply]
this didn't work for me - for some reason, the whole graph disappears

**Reply by wilkyrlx** [reply, marked as answer]
I was able to fix it by using fixed position

**Reply by faseidl** [reply]
Can you show the code that worked? (I've experimented with both fixed and relative and neither seems to work. Fixed makes the graph very tall (taller than the screen) and relative makes it disappear.)

**Reply by wilkyrlx** [reply]
For me, it was something like this:
```
<div style={{ position: "fixed", width: '75%', height: '75%'}}>
    <GraphCanvas etc... />
</div>
```

**Reply by faseidl** [reply]
Thanks! I got it to work using position relative, but the key was to specify an absolute height. Using a percentage resulted in a zero-high container.
